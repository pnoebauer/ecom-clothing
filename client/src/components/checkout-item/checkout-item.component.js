import React from 'react';
import { connect } from 'react-redux';

import {
	CheckoutItemContainer,
	ImageContainer,
	TextContainer,
	QuantityContainer,
	RemoveButtonContainer
} from './checkout-item.styles';

import { clearItemFromCart, addItemToCart, removeItemFromCart } from '../../redux/cart/cart.actions';

const CheckoutItem = ({ cartItem, clearItem, addItem, removeItem }) => {
	const {name, imageUrl, price, quantity} = cartItem;
	return (
		<CheckoutItemContainer>
			<ImageContainer>
				<img src={imageUrl} alt='item'/>
			</ImageContainer>
			<TextContainer>{name}</TextContainer>
			<QuantityContainer>
				<div onClick={() => removeItem(cartItem)}>
					&#10094;
				</div>
				<span>{quantity}</span>
				<div onClick={() => addItem(cartItem)}>
					&#10095;
				</div>
			</QuantityContainer>
			<TextContainer>${price}</TextContainer>
			<RemoveButtonContainer onClick={() => clearItem(cartItem)}>
				&#10005;
			</RemoveButtonContainer>
		</CheckoutItemContainer>
	);
}

const mapDispatchToProps = dispatch => ({
	clearItem: item => dispatch(clearItemFromCart(item)),
	addItem: item => dispatch(addItemToCart(item)),
	removeItem: item => dispatch(removeItemFromCart(item))
});

export default connect(null, mapDispatchToProps)(CheckoutItem);