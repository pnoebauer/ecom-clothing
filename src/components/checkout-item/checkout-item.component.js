import React from 'react';

import './checkout-item.styles.scss';

const CheckOutItem = () => (
	<div>
		
	</div>
);

export default CheckOutItem;