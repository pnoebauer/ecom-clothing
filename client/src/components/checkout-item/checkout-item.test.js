import React from 'react';
import {shallow} from 'enzyme';
import {CheckoutItem} from './checkout-item.component';
import {RemoveButtonContainer} from './checkout-item.styles';

describe('CheckoutItem component tests', () => {
	let wrapper;
	let mockAddItem;
	let mockClearItem;
	let mockRemoveItem;

	const mockItem = {
		imageUrl: 'www.testImage.com',
		price: 10,
		name: 'hats',
		quantity: 2,
	};

	beforeEach(() => {
		mockClearItem = jest.fn();
		mockAddItem = jest.fn();
		mockClearItem = jest.fn();
		mockRemoveItem = jest.fn();

		const mockProps = {
			cartItem: mockItem,
			clearItem: mockClearItem,
			addItem: mockAddItem,
			removeItem: mockRemoveItem,
		};

		wrapper = shallow(<CheckoutItem {...mockProps} />);
	});

	test('render CheckoutItem component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('display correct info', () => {
		const {quantity} = mockItem;

		const quantityText = wrapper.find('span').text();
		expect(quantityText).toEqual(`${quantity}`);
	});

	test('call mockAddItem when the #addItem button is clicked', () => {
		wrapper.find('#addItem').simulate('click');
		expect(mockAddItem).toHaveBeenCalled();
		expect(mockAddItem).toHaveBeenCalledWith(mockItem);
	});

	test('call mockRemoveItem when the #removeItem button is clicked', () => {
		// const result = wrapper.find(CartDropdownButton);
		// console.log(result.debug());
		wrapper.find('#removeItem').simulate('click');
		expect(mockRemoveItem).toHaveBeenCalled();
		expect(mockRemoveItem).toHaveBeenCalledWith(mockItem);
	});

	test('call mockClearItem when the RemoveButtonContainer is clicked', () => {
		// const result = wrapper.find(CartDropdownButton);
		// console.log(result.debug());
		wrapper.find(RemoveButtonContainer).simulate('click');
		expect(mockClearItem).toHaveBeenCalled();
		expect(mockClearItem).toHaveBeenCalledWith(mockItem);
	});
});
