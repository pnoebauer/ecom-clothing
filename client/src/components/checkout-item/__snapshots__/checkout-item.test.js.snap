// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component tests render CheckoutItem component 1`] = `
<checkout-itemstyles__CheckoutItemContainer>
  <checkout-itemstyles__ImageContainer>
    <img
      alt="item"
      src="www.testImage.com"
    />
  </checkout-itemstyles__ImageContainer>
  <checkout-itemstyles__TextContainer>
    hats
  </checkout-itemstyles__TextContainer>
  <checkout-itemstyles__QuantityContainer>
    <div
      id="removeItem"
      onClick={[Function]}
    >
      ❮
    </div>
    <span>
      2
    </span>
    <div
      id="addItem"
      onClick={[Function]}
    >
      ❯
    </div>
  </checkout-itemstyles__QuantityContainer>
  <checkout-itemstyles__TextContainer>
    $
    10
  </checkout-itemstyles__TextContainer>
  <checkout-itemstyles__RemoveButtonContainer
    onClick={[Function]}
  >
    ✕
  </checkout-itemstyles__RemoveButtonContainer>
</checkout-itemstyles__CheckoutItemContainer>
`;
